src/main/lombok/org/heigit/ohsome/ohsomeapi/inputprocessing/InputProcessor.java

```diff
@@ -212,11 +211,13 @@ public InputProcessor(HttpServletRequest servletRequest, boolean isSnapshot, boo
       mapRed =
           mapRed.areaOfInterest(OSHDBGeometryBuilder.boundingBoxOf(boundary.getEnvelopeInternal()));
     } else {
-      mapRed = mapRed.areaOfInterest((Geometry & Polygonal) boundary);
+      try {
+        mapRed = mapRed.areaOfInterest((Geometry & Polygonal) boundary);
+      } catch (TopologyException e) {
+        throw new BadRequestException("Invalid geometry: " + e.getMessage());
```

Comment:
maybe this error message could be slightly more specific, e.g. that the invalid geometry was encountered in the boundary parameter. Also for consistency it could be stored in the `ExceptionMessages` alongside the other error messages, right?